contributor-docs/versioning.md

```diff
@@ -0,0 +1,132 @@
+# Versioning
+
+<!-- prettier-ignore-start -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+## Table of Contents
+
+- [Overview](#overview)
+- [Changes](#changes)
+  - [Examples](#examples)
+    - [A prop is added to a component](#a-prop-is-added-to-a-component)
+    - [An existing prop is deprecated](#an-existing-prop-is-deprecated)
+    - [The DOM node that an `id` corresponds to is changed](#the-dom-node-that-an-id-corresponds-to-is-changed)
+    - [The DOM node that an `aria-label` corresponds to is changed](#the-dom-node-that-an-aria-label-corresponds-to-is-changed)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- prettier-ignore-end -->
+
+## Overview
+
+The Primer team aims to follow
+[Semantic Versioning](https://semver.org/) (semver) for each of the packages
+that we ship. From semver.org, this means that:
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+>
+> 1. **MAJOR** version when you make incompatible API changes,
+> 2. **MINOR** version when you add functionality in a backwards compatible
+>    manner, and
+> 3. **PATCH** version when you make backwards compatible bug fixes.
+>
+> _Additional labels for pre-release and build metadata are available as
+> extensions to the MAJOR.MINOR.PATCH format._
+
+As a result, whenever you see a `minor` or `patch` update for a package from the
+Primer you should feel confident that you can update without
+anything breaking in your project. For a full list of changes and their
+corresponding semver bumps, check out the [changes](#changes) table below.
+
+For a full list of releases, visit our [releases](https://github.com/primer/react/releases) page.
+
+## Changes
+
+| Category   | Type of change                                                | semver bump |
+| :--------- | :------------------------------------------------------------ | :---------- |
+| Component  | A component is added                                          | `minor`     |
+|            | A component is deprecated                                     | `major`     |
```

Comment:
This one's interesting:

1. In Primer Primitives deprecating something doesn't constitute a material breaking change in the output, but serves as a flag that something will soon be removed and that we have provided an alternative to ease migration
2. During v35 major release, I recall the team suggested in a retro that we don't make deprecations major. We had the same feedback from library users.

Reply:
@rezrah I definitely agree, deprecations in `minor` releases feel more appropriate/align with expectations.

Reply:
> During v35 major release, I recall the team suggested in a retro that we don't make deprecations major. We had the same feedback from library users.

I am very curious to learn more about this. It is very interesting.

We will be deprecating `UnderlineNav` in v36 and it is planned to be a major change. We re-wrote the component entirely and the public API has changed. I don't have much experience with versioning but I am super keen to improve! I am curious to know if it is still possible to refactor `UnderlineNav` and introduce as a minor change with warning the deprecation and providing the new API as an alternative 🤔  Would it worth giving a try or not really at this stage? 

[Current UnderlineNav](https://primer.style/react/UnderlineNav)
[Newly written UnderlineNav](https://primer.style/react/drafts/UnderlineNav2)

Reply:
@joshblack @broccolinisoup I've somehow found the FigJam file I was referencing in my earlier comment. You should receive a link to it in your emails. 

@broccolinisoup - my reading of your situation based on the diagram would be to issue the deprecation notice through comms (docs, changelogs, etc) and a patch (or minor) update of the libraries (for JSDOC `@deprecate` annotations). Main thing is to leave it in-place. We're just issuing notice AoT.

Then wait for the next major release (~couple of weeks) to move it into the deprecated bundle and move UnderlineNavV2 from drafts into main bundle. Eventually few months after, we can remove deprecated component altogether. 

Could this still work for you?

Reply:
@joshblack @broccolinisoup I've somehow found the FigJam file I was referencing in my earlier comment. You should receive a link to it in your emails. 

@broccolinisoup - my reading of your situation based on the diagram would be to issue the deprecation notice through comms (docs, changelogs, etc) and a patch (or minor) update of the libraries (for JSDOC `@deprecate` annotations). Main thing is to leave it in-place. We're just issuing notice AoT.

Then wait for the next major release (~couple of weeks) to move it into the deprecated bundle and move UnderlineNavV2 from drafts into main bundle. Eventually few months after, we can remove deprecated component altogether. 

Could this still work for you?

Reply:
> I've somehow found the FigJam file I was referencing in my earlier comment. You should receive a link to it in your emails.

Thanks, this is brilliant  🙌🏼

> Could this still work for you?

@rezrah Thank you so much for clarifying the process! According to your explanation, we are on track. The only thing is different than the plan is updating the old UnderlineNav with the `@deprecate` status as a patch (or minor) update before the major release. Would that still be a good practise without promoting the new UnderlineNav as an alternative component (As this plan to come as a major later)? I think this is what I am trying to wrap my head around.

 Thank you so much 🙏🏼

Reply:
Hi! @rezrah just wanted to resurface this, would love to hear your thoughts 😊

Reply:
That sounds great! I appreciate the very detailed explanation as well 🙇🏻‍♀️  I'll capture this info in the deprecation process documentation [PR](https://github.com/primer/react/pull/2733).